### ISSUE TITLE
ansible-galaxy ignores the -f flag

### ISSUE THREAD

**Issue by thanatos** [ansible/ansible#25933]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
`ansible-galaxy`

##### ANSIBLE VERSION
```
» ansible --version
ansible 2.3.1.0
  config file = /Users/roy/data/hc-entity-discovery/deploy/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Dec 18 2016, 07:03:39) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### OS / ENVIRONMENT
OS X

##### SUMMARY
The `-f` flag does not appear to be honored.

For example, running `ansible-galaxy` with `-f` doesn't result in roles getting re-installed:
```
» ansible-galaxy install -f -r requirements.yml
- ubuntu is already installed, skipping.
- newrelic is already installed, skipping.
- newrelic_app is already installed, skipping.
- nginx is already installed, skipping.
[ more of the same ]
```

##### EXPECTED RESULTS
`ansible-galaxy` re-downloads the roles.

##### ACTUAL RESULTS
`ansible-galaxy` didn't do anything.

This means that if the role drifts out of date (newer changes upstream) or the requirement changes, they won't be picked up.

(We've had problems where, yes, the role is already installed, but the requirement has changed. The person running the deployment is not necessarily the same person as the person who changed the requirement, and might not know if the roles need updating. Thus, we always pass `-f` to ensure up-to-date roles.)

**Comment by hvindin** [comment]
I think this was resolved here: [galaxy-issues #249](https://github.com/ansible/galaxy-issues/issues/249) and was fixed here #23391

**Comment by s-hertel** [comment]
Fixed by https://github.com/ansible/ansible/commit/78836ec0b9719a62fa0f8619707a9f411ed4a4f0 and will be in the 2.3.2 release.
